nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -348,14 +352,78 @@ export class FlowEffects {
                         this.flowService.goToRemoteProcessGroup(request);
                     } else {
                         this.store.dispatch(
-                            showOkDialog({ title: 'Remote Process Group', message: 'No target URI defined.' })
+                            FlowActions.showOkDialog({
+                                title: 'Remote Process Group',
+                                message: 'No target URI defined.'
+                            })
                         );
                     }
                 })
             ),
         { dispatch: false }
     );
 
+    startRemoteProcessGroupPolling$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.startRemoteProcessGroupPolling),
+            switchMap((request) => {
+                return interval(3000, asyncScheduler).pipe(
+                    takeUntil(this.actions$.pipe(ofType(FlowActions.stopRemoteProcessGroupPolling)))
+                );
+            }),
+            switchMap(() => {
+                return of(FlowActions.refreshRemoteProcessGroup());
+            })
+        )
+    );
+
+    requestRemoteProcessGroupPolling$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.requestRemoteProcessGroupPolling),
+            switchMap(() => {
+                return of(FlowActions.startRemoteProcessGroupPolling());
+            })
+        )
+    );
+
+    refreshRemoteProcessGroup$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.refreshRemoteProcessGroup),
+            concatLatestFrom(() => this.store.select(selectRpgToPoll)),
+            map(([, rpgToPoll]) => {
+                return rpgToPoll;
+            }),
+            isDefinedAndNotNull(),
+            switchMap((rpgToPoll) =>
+                from(
+                    this.flowService.getRemoteProcessGroup(rpgToPoll.id).pipe(
+                        map((response: any) => {
+                            const entity = response;
+
+                            if (rpgToPoll.refreshTimestamp !== response.component.flowRefreshed) {
+                                this.store.dispatch(FlowActions.stopRemoteProcessGroupPolling());
+                            } else {
+                                entity.component.flowRefreshed = 'Refreshing...';
+                            }
+
+                            return FlowActions.loadRemoteProcessGroupSuccess({
+                                response: {
+                                    id: entity.id,
+                                    remoteProcessGroup: entity
+                                }
+                            });
+                        }),
+                        catchError((errorResponse: HttpErrorResponse) => {
+                            this.store.dispatch(FlowActions.stopRemoteProcessGroupPolling());
+
+                            return of(this.errorHelper.handleLoadingError(status, errorResponse));
```

Comment:
`status` isn't defined here. I understand that error handling has not been introduced in this `FlowEffects` yet (there is a line item for it in NIFI-12400). But the `hanldingLoadingError` method was meant to handle errors loading the initial state. In addition to stopping polling, I suspect that here we would want to check if the error is one that should be shown in context and let that drive whether we want a snackbar or full screen error.

Reply:
I think this is a copy past error...